platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/offline/MergeOfflineRegionsActivity.kt

```diff
@@ -57,6 +57,27 @@ class MergeOfflineRegionsActivity : AppCompatActivity() {
     }
   }
 
+  /**
+   * Since we expect from the results of the offline merge callback to interact with the hosting activity,
+   * we need to ensure that we are not interacting with a destroyed activity.
+   */
+  private class MergeCallback(private var activityCallback: OfflineManager.MergeOfflineRegionsCallback?) : OfflineManager.MergeOfflineRegionsCallback {
```

Comment:
Why not use a weak reference here? this would avoid the whole

Reply:
https://github.com/mapbox/mapbox-gl-native/pull/14601#discussion_r281786298

WeakReference causes races with lifecycle events. It does prevents leaks, that's good, and also keeps a whole new category of potential errors.